src/App.tsx

```diff
@@ -1,27 +1,64 @@
-import React from 'react';
+import { FC, useRef, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, getSomeGoods, getGoodsSortedByColor } from './api/goods';
+import { Good, QueryType } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const quaryType = useRef<QueryType | null>(null);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const getAllGoods = async () => {
+    if (quaryType.current !== QueryType.ALL) {
+      setGoods(await getAll());
+      quaryType.current = QueryType.ALL;
+    }
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const getFiveGoods = async () => {
+    if (quaryType.current !== QueryType.SOME) {
+      setGoods(await getSomeGoods(5));
+      quaryType.current = QueryType.SOME;
+    }
+  };
```

Comment:
I think if we invert condition it will improve code readability(optional change)

```suggestion
  const getFiveGoods = async () => {
    if (quaryType.current === QueryType.SOME) {
      return;
    }
    
    setGoods(await getSomeGoods(5));
    quaryType.current = QueryType.SOME;
  };
```